lib/api/apiUtils/bucket/bucketWebsite.js

```diff
@@ -0,0 +1,275 @@
+                        return errors.InvalidArgument
+                            .customizeDescription(errMsg);
+                    }
+                    const objKey = elem.charAt(0).toLowerCase() + elem.slice(1);
+                    redirectObj[objKey] = value;
+                }
+            }
+            if (_xmlContainsElem(rule.Redirect, 'ReplaceKeyPrefixWith')
+            && _xmlContainsElem(rule.Redirect, 'ReplaceKeyWith')) {
+                errMsg = 'Redirect must not contain both ReplaceKeyWith ' +
+                'and ReplaceKeyPrefixWith';
+                return errors.InvalidRequest.customizeDescription(errMsg);
+            }
+            routingRule.setRedirect(redirectObj);
+            if (Array.isArray(rule.Condition) && rule.Condition.length === 1) {
+                const conditionObj = {};
+                if (!_xmlContainsElem(rule.Condition, 'KeyPrefixEquals') &&
+                !_xmlContainsElem(rule.Condition,
+                'HttpErrorCodeReturnedEquals')) {
+                    errMsg = 'Condition is not well-formed. ' +
+                    'Condition should contain valid KeyPrefixEquals or ' +
+                    'HttpErrorCodeReturnEquals element.';
+                    return errors.InvalidRequest.customizeDescription(errMsg);
+                }
+                if (rule.Condition[0].KeyPrefixEquals) {
+                    const value = rule.Condition[0].KeyPrefixEquals[0];
+                    if (typeof value !== 'string' || value === '') {
+                        errMsg = 'Condition KeyPrefixEquals is not well-formed';
+                        return errors.InvalidArgument
+                            .customizeDescription(errMsg);
+                    }
+                    conditionObj.keyPrefixEquals = value;
+                }
+                if (rule.Condition[0].HttpErrorCodeReturnedEquals) {
+                    const code = parseInt(rule.Condition[0]
+                        .HttpErrorCodeReturnedEquals[0], 10);
+                    if (isNaN(code)) {
+                        errMsg = 'The provided HTTP error code is not valid. ' +
+                        'It should be a string containing a number.';
+                        return errors.MalformedXML.customizeDescription(errMsg);
+                    }
+                    if (!(code > 399 && code < 600)) {
+                        errMsg = `The provided HTTP error code (${code}) is ` +
+                        'not valid. Valid codes are 4XX or 5XX.';
+                        return errors.InvalidRequest
+                            .customizeDescription(errMsg);
+                    }
+                    conditionObj.httpErrorCodeReturnedEquals = code;
+                }
+                routingRule.setCondition(conditionObj);
+            }
+            websiteConfig.addRoutingRule(routingRule);
+        }
+    }
+    return websiteConfig;
+}
+
+export function parseWebsiteConfigXml(xml, log, next) {
```

Comment:
I understand that this function is used along with `async.waterfall`, but I think something like `cb` would be a  clearer parameter name in this context.